themes/hive/layouts/partials/footer.html

```diff
@@ -0,0 +1,55 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License. -->
+
+<footer class="black-background static-bottom" style="padding:30px">
+    <div class="row">
+        <div class="col-3">
+            <a href="{{ .Site.Params.apache.apacheURL }}">
+                <img src="{{.Site.BaseURL}}{{ .Site.Params.logo }}" width="270" height="100"  alt="Apache Software Foundation"></a>
+            </a>
+        </div>
+        <div class="col-9">
+            <p class="footer-text">Apache is a non-profit organization helping open-source
+                software projects released under the Apache
+                <a href="{{ .Site.Params.apache.license }}">license</a>
+                and managed with
+                <a href="{{ .Site.Params.apache.howItWorks }}">
+                    open governance</a> and
+                <a href="{{ .Site.Params.apache.privacyPolicy }}">
+                    privacy policy</a>. See upcoming
+                <a href="{{ .Site.Params.apache.apacheEvents }}">Apache Events</a>.
+                If you discover any
+                <a href="{{ .Site.Params.apache.security }}">security</a> vulnerabilities, please
+                report them privately. Finally,
+                <a href="{{ .Site.Params.apache.sponsors }}">thanks
+                </a> to the sponsors who
+                <a href="{{ .Site.Params.apache.donate }}">
+                    donate</a> to the Apache Foundation.
+            </p>
+        </div>
+
+
+    </div>
+    <div class="copyright row">
+        <a href="{{ .Site.BaseURL }}" style="color:grey">
+            {{ with .Site.Params.Apache.Copyright | default .Site.Title }} {{ . | safeHTML }} {{ end }}
+        </a>
+    </div>
+
+</footer>
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
```

Comment:
Is it safe to use which encryption algorithm is used in the HTML?

Reply:
Yes, it is safe. 
Bootstrap is included via CDN from the official source mentioned here: https://getbootstrap.com/ 
It looks like it is encrypted using SHA-384 which belongs to the same family as SHA-2. 
SHA-384 is supposed to be a faster hashing algorithm on 64-bit machines.